### ISSUE TITLE
RuntimeError: Could not find mono_mz in theoretical env!

### ISSUE THREAD

**Issue by mengchen18** [weerapana-lab/envoMatch#4]
Hi @ajmaurais,

This issue follows the discussion here:

https://github.com/weerapana-lab/ionFinder/issues/9#issuecomment-1524221935

The data to reproduce the error message can still be downloaded here:

https://syncandshare.lrz.de/getlink/fiJkKpudQ7rH22k9xNWGgm/ionFinder_error_reproduce

I have a encountered another problem with the recent version that fixed the warnings (commit 66031cf). The older version works fine (except some warnings). 

What I did is to download and build the most recent version of envoMatch. Then run:

```
envoMatch -a ~/code/envoMatch/db/atom_tables/cit_diff_mod_atoms.txt --parallel 0 peptide_cit_stats.tsv
```

The output including error message is:

```
Reading ms1 files using 1 thread(s)...
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [01:39<00:00, 99.37s/it]

Searching for envelopes using 1 thread(s)...
  2%|████▍                                                                                                                                                                                    | 291/12096 [00:43<46:04,  4.27it/s]/home/chen/.local/lib/python3.8/site-packages/envoMatch/modules/consensusEnvelope.py:205: RuntimeWarning: invalid value encountered in double_scalars
  self._actual[i].point.int /= max_int
/home/chen/.local/lib/python3.8/site-packages/envoMatch/modules/consensusEnvelope.py:205: RuntimeWarning: divide by zero encountered in double_scalars
  self._actual[i].point.int /= max_int

...


50%|███████████████████████████████████████████████████████████████████████████████████████████▍                                                                                            | 6007/12096 [14:44<08:11, 12.39it/s]/home/chen/.local/lib/python3.8/site-packages/envoMatch/modules/consensusEnvelope.py:205: RuntimeWarning: invalid value encountered in double_scalars
  self._actual[i].point.int /= max_int
/home/chen/.local/lib/python3.8/site-packages/envoMatch/modules/consensusEnvelope.py:205: RuntimeWarning: divide by zero encountered in double_scalars
  self._actual[i].point.int /= max_int
 53%|████████████████████████████████████████████████████████████████████████████████████████████████▌                                                                                       | 6351/12096 [15:35<12:42,  7.54it/s]/home/chen/.local/lib/python3.8/site-packages/envoMatch/modules/consensusEnvelope.py:205: RuntimeWarning: invalid value encountered in double_scalars
  self._actual[i].point.int /= max_int
/home/chen/.local/lib/python3.8/site-packages/envoMatch/modules/consensusEnvelope.py:205: RuntimeWarning: divide by zero encountered in double_scalars
  self._actual[i].point.int /= max_int
 55%|████████████████████████████████████████████████████████████████████████████████████████████████████▍                                                                                   | 6601/12096 [16:10<13:27,  6.80it/s]
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/home/chen/.local/lib/python3.8/site-packages/envoMatch/main.py", line 137, in _annotate_ms1
    consensus[s].set_mono(mono_mz=mono_mzs[s])
  File "/home/chen/.local/lib/python3.8/site-packages/envoMatch/modules/consensusEnvelope.py", line 129, in set_mono
    raise RuntimeError('Could not find mono_mz in theoretical env!')
RuntimeError: Could not find mono_mz in theoretical env!
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/chen/.local/bin/envoMatch", line 8, in <module>
    sys.exit(main())
  File "/home/chen/.local/lib/python3.8/site-packages/envoMatch/main.py", line 257, in main
    env_data = list(tqdm(pool.imap(functools.partial(_annotate_ms1,
  File "/home/chen/.local/lib/python3.8/site-packages/tqdm/std.py", line 1178, in __iter__
    for obj in iterable:
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 868, in next
    raise value
RuntimeError: Could not find mono_mz in theoretical env!
```

Again, thanks for your help. 
Chen

**Comment by ajmaurais** [comment]
One problem is that your ms1 spectra in your mzML file are in profile mode.  `envoMatch` will work best if they are centroided. I used MSconvert with vendor peak picking for MS levels 1 and 2 to make a mzML file which I used for further testing. The peptide that was causing this error was an unmodified peptide which, according to MaxQuant, has a +6 charge state. I just committed a fix so your msms_parsed file should now work. However, I would recommend running `ionFinder` with the `-m 0` option to only include citrullinated peptides in the peptide_cit_stats report. Filtering out unmodified peptides will substantially decrease the time it takes `envoMatch` to run.
